### ISSUE TITLE
mxnet model successfully converted to ncnn, segmentation fault during inference (with no error message)

### ISSUE THREAD

**Issue by cyrusbehr** [Tencent/ncnn#1292]
I successfully converted an MXNET model to ncnn using the `mxnet2ncnn` utility. During the conversion, there no were errors (aside from the bug discussed in [this](https://github.com/Tencent/ncnn/issues/1291) issue). 

The first layer of my network is called `/input_11` and has a data shape of `(3,24,24)`. Note that the MXNET model expects a batch size of 64 (not sure if this is related to the error I am experiencing).
The output layer is called `activation1`  and is a Sigmoid layer.

I am trying to run inference using the following code:
```
    cv::Mat data = cv::imread(imagePath);
    cv::resize(data, data, cv::Size(24, 24));
    ncnn::Mat in = ncnn::Mat::from_pixels(data.data, ncnn::Mat::PIXEL_RGB, data.cols, data.rows);
    ncnn::Extractor ex = net.create_extractor();
    ex.input("/input_11", in);
    ncnn::Mat out;
    ex.extract("activation1", out);
```
On the call to `ex.extract()` there is a segmentation fault. The only output is as follows, and nothing else:
```
Segmentation fault (core dumped)
```

I have confirmed that the original model does work when loaded with MXNET. Do you have any advice?

**Comment by cyrusbehr** [comment]
More information:
After doing some debugging, it looks like the segfault is occurring on this line [here](https://github.com/Tencent/ncnn/blob/48e3e7d49cb203a9a83fa0304cceaf4874662ece/src/net.cpp#L1288)
For the layer at index 2, `layer->bottoms` has size zero, thus causing a segfault. Not sure why this is the case

**Comment by bindog** [comment]
Do you have any progress now?
You need to provide more information to identify where the problem is, like the structure of your network, the content of the param file and which type the layer is and the layer before it when the first `layer->bottoms.size() = 0` occur

**Comment by cyrusbehr** [comment]
Looks like the issue had to do with the fact that our model was created in Keras then exported to MXNET (before being converted to NCNN).

**Comment by rcastill** [comment]
Came into this error while testing `mnet.25-opt`. After a lot of debugging (and confirming is the exact same case as @cyrusbehr described), I changed the version of ncnn from [90f8359](https://github.com/Tencent/ncnn/commit/90f8359961b8d323a8ea4136430d68d65d5e35a4) to older [9593783](https://github.com/Tencent/ncnn/commit/9593783be34f03e6ca45c524b6176748768ef472) and **it worked**.

I hope you can find use to this.

**Comment by cbecker** [comment]
@bindog 
[relumodel.zip](https://github.com/Tencent/ncnn/files/4236783/relumodel.zip)
I found the same issue with the latest release (20200106).
This is a network that does ReLU then multiply by a constant. Attached the onnx file and the pytorch code that produces it.

**Comment by cbecker** [comment]
(and also reverting back to v20191113 fixes the isue)

**Comment by cbecker** [comment]
I think I found what's the error. I'll open another bug report as it may not be related to this one: https://github.com/Tencent/ncnn/issues/1543

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
